upt_macports/upt_macports.py

```diff
@@ -82,6 +118,10 @@ def _pkgname(self):
     def _normalized_macports_name(name):
         return f'py-{name}'
 
+    @staticmethod
+    def _normalized_macports_folder(name):
+        name = name.lower()
+        return f'py-{name}'
```

Comment:
this is not correct; the directory name should be the same as the ```field``` name in the Portfile. There we decided to follow the upstream name, including possible capitalization

Reply:
I agree that those two should match (and I would use a function to construct the name here), but I still don't like the idea of native capitalisation, in particular because it may differ on github and pypi.

I'm not sure if devs would have the same opinion after seeing a subset of PR camelcasing names.

Reply:
Using a function here would indeed make sense. I don't understand your comment about possible differences between GitHub and PyPI though, can you elaborate on that? From what I have seen pretty much anywhere people use whatever name "upstream" has decided to call the package (including capitalization).

The only people who responded on the mailing list to this topic were Rainer, Josh, and Ryan - and from what I gathered from there is that "following upstream would be fine". So either other devs don't care or have no opinion... But again, I really don't have a strong preference and it seems there are currently only 12 ports that do have capitalization, so I am fine with going to only lower case as well. That would be more consistent with the Perl packages - we just need to make an executive decision now and move on ;)

Reply:
I just did a quick analysis of pypi package names and in that 20108/183646 have some type of capitalization which is around 10%.

Seeing this and the fact that I have never cared about the capitalization while installing packages via pip, I think we should follow lower case and this will also be consistent with perl packages as point be @reneeotten

Reply:
> I just did a quick analysis of pypi package names and in that 20108/183646 have some type of capitalization which is around 10%.

This might be something worth mentioning in the discussion about the issue.
 
> Seeing this and the fact that I have never cared about the capitalization while installing packages via pip, I think we should follow lower case and this will also be consistent with perl packages as point be @reneeotten

That makes at least three of us :)

What I meant with potentially different capitalisation was also mention in my response to your query. If we had a package `py-macports`: how would you capitalise it? Note that all of our repository names are lowercase, yet our official name is CamelCase (MacPorts).

We seem to have just 12 ports starting with uppercase. I checked a random one, `py-SDL2`, and that one contains
```tcl
github.setup        marcusva py-sdl2 ...
```
If this wasn't a python module, we tend to use the exact same name as the github repository, which in this case is `py-sdl2` (lowercase), which could be a hint that we might be using "wrong" capitalisation, and if you look at `setup.py` you'll see:
```python
       "url": "https://github.com/marcusva/py-sdl2",
        "download_url": "https://pypi.python.org/pypi/PySDL2",
```
Now: which capitalisation is "correct"? If you check the contents of the package, it's mostly lowercase. Also, it's
```python
from sdl2 import ...
```
so lowercase when used as a python module name.

Another confusion of course is whether this should be `py-pysdl2` or `py-sdl2`, but that's yet a different topic.

If MacPorts is using a roughly representative sample of `pypi`, this means that we would need to rename roughly 150 ports, and also fix all ports which have those ports declared as dependencies.

The following comment is taken out of context, but I would prefer if we used something like
```tcl
pypi.setup upt 0.7
```
which would automatically set name, version, python.rootname, master_sites, distname etc. Similar to how little the perl modules need.

Reply:
okay, let's then move to lowercase names for Python ports and let's stick with this decision at least during the remainder of the GSoC ;) 

@Korusuke please open another PR to make this change and keep in mind the discussion in e-mails we had about the need to set ```python.rootname``` to the upstream name if it differs from the MacPorts package name. That's the easiest way to keep the remainder of the Python template the same. 

@mojca I would consider suggestions about changing the Python PortGroup as outside of the scope of the ```upt``` project. Nevertheless, accommodating changes like that at a later stage in the template should be easy enough.

Reply:
> okay, let's then move to lowercase names for Python ports and let's stick with this decision at least during the remainder of the GSoC ;)
> 
> @Korusuke please open another PR to make this change and keep in mind the discussion in e-mails we had about the need to set `python.rootname` to the upstream name if it differs from the MacPorts package name. That's the easiest way to keep the remainder of the Python template the same.

I don't get why would we need to set `python.rootname` as from what I understand it's only used if the package name is different from MacPorts name except for capitalization but that won't be the case for upt ever

Reply:
Please take a look at, for example, ```py-sphinx```. This is all lowercase in MacPorts, but upstream/PyPI uses ```Sphinx```. In this case for MacPorts the name is ```py-sphinx```, and ```python.rootname``` is set to ```sphinx``` by the Python PortGroup. Now, if you look in our python.Portfile template you will notice that this variable is used for things like ```master_sites``` and ```distname```, which are now not correct anymore because of the capitalization. At least I am pretty sure that's the case, but please verify this yourself by editing that Portfile and replacing all occurrences of "Sphinx" with "sphinx" in the master_sites and distname as that's the result you would get with upt.

In cases like this you can either set ```python.rootname``` to mach upstream, in the above example to ```Sphinx``` and leave the rest of the template untouched. Alternatively, you would need to change the ```distname``` and ```master_sites``` as is currently done in the ```py-sphinx``` Portfile. I think the easiest option is easier.

Reply:
Okay Thanks for the explanation, I thought capitalization won't matter as it doesn't matter with urls but forgot that while downloading its actually a directory structure.

Added a PR as per this but while trying it out it somewhat works....more description in the PR.

Reply:
There are a lot of existing ports not using python.rootname simply because it's a relatively recent addition.